docs/content/advanced/backend/events/events-architecture.md

```diff
@@ -0,0 +1,85 @@
+
+The signature of each method and an example can be seen below:
+
+**Emit**
+```ts
+// signature
+async emit<T>(eventName: string, data: T, options: Record<string, unknown> & EmitOptions = { attempts: 1 }): Promise<StagedJob | void>
+
+// example
+eventBusService.emit("product.created", { id: "prod_id" }, { attempts: 2 })
+```
+In the example above, the `EventBusService` emits event `product.created`, that contains the id of a product. Additionally, the `options` argument is used to configure the number of times a job is run - default is one - which kicks in  in case a subscriber fails to process it - here pass `attemps: 2` which means it is retried once. More on that later.
+
+**Subscribe**
+```ts
+// signature
+subscribe(event: string | symbol, subscriber: Subscriber, context: SubscriberContext): this
+
+// example
+eventBusService.subscribe("product.created", (data) => console.log(data.id))
+```
+In the example above, the `EventBusService` subscribes to event `product.created` and writes the id to the server log upon processing it. The method takes a third optional argument, `context`, that allows you to configure the id of the subscriber. More on that later.
+
+Learn more about [subscribers and see concrete implementations](https://docs.medusajs.com/advanced/backend/subscribers/create-subscriber). 
+
+### Processing
+In the `EventBusService` service, you'll find the `worker_` method, which defines the logic run on each job in the queue. By default, all subscribers to that event (job) are retrieved and for each of the them, the stored job data - provided in emit - is passed to them.
+
+**Retrying**
+A subscriber might fail to process an event. This could happen because the subscriber communicates with a third party service currently down or due an error in the logic of the subscriber. In some cases, you might want to retry those failed subscribers. 
+
+As briefly described previously, you can pass options when emitting an event, that are used to configure how the queue worker processes your job. If you pass  `attempts` upon emitting the event, the processing of the event is retried. 
+
+Aside from `attempts`, there are other options to futher configure the retry mechanism:
+
+IMPORTANT: If you have more than one subscriber attached to a single event, you are required to define subscriber ids that allows the `EventBusService` to differentiate between them when retrying the job-processing. Otherwise, **all** subscribers are run again, which can lead to data inconsistencies or general unwanted behavior in your system. Conversely, you might want all subscribers to run on every retry; in that case, the ids are redundant. 
```

Comment:
Suggestion: not only when multiple subscribers are attached but also if multiple instances are running. Otherwise, if another instance run the retry, it will re run already successful handkers